realm/realm-library/src/androidTest/kotlin/io/realm/ParameterizedDictionaryTests.kt

```diff
@@ -0,0 +1,371 @@
+ * [RealmModel] and [Mixed] (and in turn all possible types supported by Mixed) in both `managed`
+ * and `unmanaged` modes.
+ *
+ * In order to streamline the testing for managed dictionaries we use Kotlin's reflection API
+ * `KFunction1` and `KFunction2`. These two methods provide access to the Java getters and setters
+ * used to work with each dictionary field.
+ */
+@RunWith(Parameterized::class)
+class ParameterizedDictionaryTests(
+        private val tester: DictionaryTester
+) {
+
+    /**
+     * Initializer for parameterized tests.
+     */
+    companion object {
+
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0}")
+        fun testTypes(): List<DictionaryTester> {
+            return DictionaryMode.values().map { type ->
+                when (type) {
+                    DictionaryMode.UNMANAGED -> unmanagedFactory()
+                    DictionaryMode.MANAGED -> managedFactory()
+                }
+            }.flatten()
+        }
+    }
+
+    @Rule
+    @JvmField
+    val configFactory = TestRealmConfigurationFactory()
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().context)
+        tester.setUp(configFactory.createConfiguration())
+    }
+
+    @After
+    fun tearDown() {
+        tester.tearDown()
+    }
+
+    @Test
+    fun constructorWithAnotherMap() {
+        tester.constructorWithAnotherMap()
+    }
+
+    @Test
+    fun isManaged() {
+        tester.isManaged()
+    }
+
+    @Test
+    fun isValid() {
+        tester.isValid()
+    }
+
+    @Test
+    fun isFrozen() {
+        tester.isFrozen()
+    }
+
+    @Test
+    fun size() {
+        tester.size()
+    }
+
+    @Test
+    fun isEmpty() {
+        tester.isEmpty()
+    }
+
+    @Test
+    fun containsKey() {
+        tester.containsKey()
+    }
+
+    @Test
+    fun containsValue() {
+        tester.containsValue()
+    }
+
+    @Test
+    fun get() {
+        tester.get()
+    }
+
+    @Test
+    fun put() {
+        tester.put()
+    }
+
+    @Test
+    fun remove() {
+        tester.remove()
+    }
+
+    @Test
+    fun putAll() {
+        tester.putAll()
+    }
+
+    @Test
+    fun clear() {
+        tester.clear()
+    }
+
+    @Test
+    fun keySet() {
+        tester.keySet()
+    }
+
+    @Test
+    fun values() {
+        tester.values()
+    }
+
+    @Test
+    fun entrySet() {
+        tester.entrySet()
+    }
+
+    @Test
+    fun freeze() {
+        tester.freeze()
+    }
+
+    @Test
+    fun copyToRealm() {
+        tester.copyToRealm()
+    }
+
+    @Test
+    fun copyFromRealm() {
+        tester.copyFromRealm()
+    }
+}
+
+/**
+ * Modes for dictionaries.
+ */
+enum class DictionaryMode {
+    UNMANAGED, MANAGED
+}
+
+/**
+ * Supported types by dictionaries. Notice that Mixed dictionaries can in turn support all these
+ * types internally (except Mixed itself).
+ *
+ * Add new types ad-hoc here.
+ */
+enum class DictionarySupportedType {
```

Comment:
I think we should rather iterate our tests over an _all possible types_ enumeration and define a utility method `isDictionarySupported` (possible on the enum itself) that returns true for the below similar ones. Then it is way easier to make tests that fails it we introduce a new type.

Reply:
I have created https://github.com/realm/realm-java/issues/7331